packages/umi-rpc-web3js/src/createWeb3JsRpc.ts

```diff
@@ -341,6 +349,37 @@ export function createWeb3JsRpc(
     }
   };
 
+  const simulateTransaction = async (
+    transaction: Transaction,
+    options: RpcSimulateTransactionOptions = {}
+  ): Promise<RpcSimulateTransactionResult> => {
+    try {
+      const tx = toWeb3JsTransaction(transaction);
+      const result = await getConnection().simulateTransaction(tx, {
+        sigVerify: options.verifySignatures,
+        accounts: {
+          addresses: options.accounts || [],
+          encoding: 'base64',
+        },
+      });
+      return {
+        err: result.value.err,
```

Comment:
nit: if you aren't specifically trying to filter out for only these 4 fields, you can just `return result.value` or if you are worreid about referencing internal points, you can return a destructured copy `return  {...result.value}`

Reply:
Here's the originaly return type form web3js

```
type SimulatedTransactionResponse = {
    err: TransactionError | string | null;
    logs: Array<string> | null;
    accounts?: (SimulatedTransactionAccountInfo | null)[] | null;
    unitsConsumed?: number;
    returnData?: TransactionReturnData | null;
  };
  ```
  
We are dropping `returnData` which is why the spread operator isn't there currently. I was building off Keith's original commit which had accounts and return data missing from the return. I can add returnData if we think its worth it.
 
 https://solana-labs.github.io/solana-web3.js/types/TransactionReturnData.html

Reply:
I've added the full return type in the end.

You'll see a spread return in the next commit.